### ISSUE TITLE
Lua borks with "out of memory" error on load

### ISSUE THREAD

**Issue by ensisoft** [emscripten-core/emscripten#22717]
Hello,

so I'm trying to migrate from Emscripten 3.1.10 to 3.1.50 so that I could have mimalloc enabled.

Trying move forward the upgrade to version 3.1.14 causes a mysterious failure.

My project has Lua embedded and built into a static library that is linked to the main executable.

Basically something like 

```
add_library(lua, 
  lua/src/lapi.c
  lua/src/lauxlib.c
  ...
)

add_executable(GameEngine,
  engine/foo.cpp
  engine/bar.cpp
  ...)

target_link_libraries(engine, lua)
```

The emscripten build generates the GameEngine.wasm, GameEngine.js, GameEngine.worker.js files.

Whenever I load the project through the HTML file (served over a python based web server) the load borks immediately with just one error printed to the web console "lua: out of memory". 

The interesting part is that I have debug logs and it never even gets as far as actually creating any Lua state object (which I'm newing). 

This issue seems very cryptic, almost smells like some kind of static construction happening before any of my function even runs and then already borking.  I'm quite confident this issue has nothing to do with "memory" per se, except perhaps maybe some kind of initialization order issue between some Lua random static state and whatever memory allocator state exists in Emscripten.

I've looked over the  change log for 3.1.14 but it doesn't give me any clues.  

Can anyone offer any help or ideas as to what might be wrong?  Thanks

**Comment by sbc100** [comment]
That seem odd since `ABORTING_MALLOC` is enabled by default, which means that if you really are out of memory the whole program will abort.   Can you track down where that message is coming from and why it thinks it is out of memory?

**Comment by ensisoft** [comment]
Thanks for the reply, after running around some time I discovered that there was in fact a "int main()" function in the Lua sources and that source file (along with that main) got built into the library.  This has been all very silent, so I had no idea this was going on.

So apparently something changed causing the linker to arrange symbols (or something similar) slightly differently and now the main function that was getting called was not "my" main function but the main function coming from the Lua library.

Closing this bug, leaving the comment here for posterity in case anyone stumbles on the same issue.
